unityfiles/Assets/Scripts/Game/Weapon/GunController.cs

```diff
@@ -144,21 +160,24 @@ private void GenerateBulletBlueprint()
 
         bullet.SetActive(false);
     }
-    // -- unity -- //
 
-    
-    void Start()
-    {   
-        //TODO: Discuss system for timer naming conventions
-        this.fireRateTimer.Set("fRate", bulletWaitTime);
-        this.GenerateBulletBlueprint();
+
+    /// <summary>
+    /// Calculates and sets the bullet wait time
+    /// </summary>
+    /// <param name="fireRate"> the new fire rate</param>
+    private void SetBulletWaitTime(float fireRate){
+        this.bulletWaitTime = 1000/fireRate;
     }
-    
 
-    void Update()
-    {
+
+    /// <summary>
+    /// Checks if it is time to fire if it is, it wil fire if not not 
+    /// </summary>
+    private void MaybeFire(){

```

Comment:
Kanskje ha denne som en bool; private boolean CanFire() som sjekker om det er tid for å skyte; også ha selve skyte mekanismen i egen funksjon.
Gjør det mulig å lettere skyte en kule uten å være avhengig av timer opplegget ved f.eks testing. Bare ha en funksjon som er Fire eller Shoot.

Reply:
Enn med og segmentere ut selve skudd fungsjonen som  kalles fra mabye fire men kan også kalles fra andre steder under utvikling